pilot/pkg/model/push_context.go

```diff
@@ -1267,6 +1267,9 @@ func (ps *PushContext) updateContext(
 			telemetryChanged = true
 		case kind.ProxyConfig:
 			proxyConfigsChanged = true
+		case kind.Namespace:
+			gatewayAPIChanged = true
+			gatewayChanged = true
```

Comment:
I generally like this change, but its pretty misleading to use `Namespace` here... We are also missing L1264 (VS change)

Reply:
it is very confusing gateway api change should be related to namespace

Reply:
Agreed, have updated the kind to `KubernetesGateway`